src/participant/Main.java

```diff
@@ -16,14 +16,14 @@ public static void main ( String[] args ) {
         for (Participant participant : participants) {
             boolean passedAllObstacles = true;
             for (Obstacle obstacle : obstacles) {
-                if ( obstacle.getDifficulty () > 0 ) {
-                    if ( obstacle instanceof Treadmill ) {
+                if (obstacle.getDifficulty () > 0) {
+                    if (obstacle instanceof Treadmill) {
```

Comment:
What if we have 100 different obstacles? Are you going to check all the 100 types?) Neither good nor flexible. According to the task description, you were supposed to create a method `overcome` that can handle such a case. Please read the description carefully and update your code.